lib/kafka/compression.rb

```diff
@@ -0,0 +1,23 @@
+require "kafka/snappy_codec"
+require "kafka/gzip_codec"
+
+module Kafka
+  module Compression
+    def self.find_codec(name)
+      case name
+      when nil then nil
+      when :snappy then SnappyCodec.new
+      when :gzip then GzipCodec.new
```

Comment:
wouldn't it be smarter to store an array of codecs like that:

```
CODECS = [
  { snappy: SnappyCodec },
  { gzip: GzipCodec }
].freeze
```
and then if nil - nothing, else if found new, otherwise fail. WOuld be easier for this and find_codec_by_id as well